### PR TITLE
Bug/remove empty on close

### PR DESCRIPTION
**Pull request by snorrees**
### Description

Restores v2 behavior: "delete on close" for empty array items.

Simply checks if the current value only contains _key and _type and if so, invokes onRemove.

Note: not tested with renderItem as it seems to be unimplemented atm?

### What to review

All the changes ;)

### Notes for release

Empty array items are removed when the edit dialog is closed.

### CODE REVIEW

File: dev/test-studio/structure/resolveStructure.ts [hunk 1]
```diff
@@ -357,7 +357,8 @@ export const structure: StructureResolver = (S, {schema}) => {
           !STANDARD_PORTABLE_TEXT_INPUT_TYPES.includes(id) &&
           !PLUGIN_INPUT_TYPES.includes(id) &&
           !EXTERNAL_PLUGIN_INPUT_TYPES.includes(id) &&
-          !DEBUG_FIELD_GROUP_TYPES.includes(id)
+          !DEBUG_FIELD_GROUP_TYPES.includes(id) &&
+          !typesInOptionGroup(S, schema, 'v3').includes(id)
```
Comment (snorrees) [inline dev/test-studio/structure/resolveStructure.ts:361]: Drive-by fix: v3 schemas was added twice in the test-studio structure.

File: packages/sanity/src/form/members/items/ArrayOfObjectsItem.tsx [hunk 2]
```diff
@@ -122,8 +142,11 @@ export function ArrayOfObjectsItem(props: MemberItemProps) {
   }, [onPathOpen, member.item.path])
 
   const handleClose = useCallback(() => {
+    if (!member.item.value || isEmpty(member.item.value)) {
+      onRemove()
+    }
     onPathOpen(member.item.path.slice(0, -1))
-  }, [onPathOpen, member.item.path])
+  }, [onPathOpen, member.item.path, member.item.value, onRemove])
```
Comment (snorrees) [inline packages/sanity/src/form/members/items/ArrayOfObjectsItem.tsx:149]: The callback will change every time value changes; I couldnt think of a way to avoid that.

File: packages/sanity/src/form/members/items/ArrayOfObjectsItem.tsx [hunk 3]
```diff
@@ -50,6 +51,25 @@ export function ArrayOfObjectsItem(props: MemberItemProps) {
     }
   })
 
+  const onRemove = useCallback(() => {
```
Comment (snorrees) [inline packages/sanity/src/form/members/items/ArrayOfObjectsItem.tsx:54]: Had to move onRemove up since it is now used by handleClose. 
Also moved onInsert since they belong together imo.

File: packages/sanity/src/form/members/items/ArrayOfObjectsItem.tsx (hunk 2 above)
Comment (mariuslundgard) [reply packages/sanity/src/form/members/items/ArrayOfObjectsItem.tsx:149]: Nitpick: It might not be worth it, but we could add a variable named `isEmptyValue` or something to avoid the callback change as often.

```ts
const isEmptyValue = !member.item.value || isEmpty(member.item.value)

const handleClose = useCallback(() => {
  if (isEmptyValue) {
    onRemove()
    onPathOpen(member.item.path.slice(0, -1))
  }
}, [isEmptyValue, member.item.path, onPathOpen])
```

**Review by mariuslundgard — APPROVED** [review]
Looks good to me!

File: packages/sanity/src/form/members/items/ArrayOfObjectsItem.tsx (hunk 2 above)
Comment (snorrees) [reply packages/sanity/src/form/members/items/ArrayOfObjectsItem.tsx:149]: Love the idea!
